### ISSUE TITLE
Find an ECT to add to your school

### ISSUE THREAD

**Issue by mason-emily** [DFE-Digital/register-early-career-teachers#473]
## Description
This story will deliver the functionality to find an ECT record in TRS that matches a TRN and date of birth entered by a school user. It will also involve playing back the name associated with the TRS record so that the user can confirm the right person has been found. 

## User story

**As** a school user
**I need** to find an ECT's record in TRS
**So that** I can continue the journey of registering the right ECT for training at my school

## Prototype, screenshots & content
[Link to prototype for 'Add an ECT' journey ](https://ecf2-prototy-v3-oz6554dztlkwtv.herokuapp.com/v3/school/home/ects) 
Password: review

[Link to 'Find an ECT' page](https://ecf2-prototy-v3-oz6554dztlkwtv.herokuapp.com/v3/school/find-ect)

Screenshots of relevant screens:

![Image](https://github.com/user-attachments/assets/20f13784-82e2-48ba-a49f-663d54275bf0)

Page content should be as per the prototype.
URL and validation messages should be as per [this doc](https://educationgovuk.sharepoint.com/:x:/r/sites/TeacherServices/Shared%20Documents/Teacher%20Continuing%20Professional%20Development/Teacher%20CPD%20Team/Register%20early%20career%20teachers/Beta/Design/Register%20ECTs%20page%20titles%20and%20URLs.xlsx?d=w0c1578ea643147b29357c7f610ee178f&csf=1&web=1&e=U8iIdi).

## Acceptance criteria

- [x] 1. When user clicks 'Continue' on the ['What you'll need to register a new ECT' page](https://github.com/DFE-Digital/register-early-career-teachers/issues/468#430), they will be navigated to the 'Find an ECT' page.
- [x] 2. User can enter a TRN and date of birth. There will be no validation for now as this will be added in a separate later ticket.
- [x] 3. When user clicks 'Continue' on the 'Find an ECT' page, the TRN and DoB entered will be used to call the TRS API to check for a match with a teacher record. Note, a similar page has been built for the AB portal.
- [x]       3a. If there is a match, user will be navigated to the 'Check if this is your ECT' page.
- [ ]       3b. If there is no match or any other error with the call, user will not be able to continue past the 'Find an ECT' page. Note, we will update the journey for when a user can't find a match or if there is an error in a separate later ticket.
- [ ] 4. On the 'Check if this is your ECT' page (when there **is** a TRS match), user will be shown the first and last name associated with the matched TRS record, as well as the TRN and date of birth. Note, there is also a middle name in TRS which we have agreed not to show. 
- [ ] 5. At the bottom of the 'Check if this is your ECT' page, there will be a link to go back and 'check details' which will navigate the user back to the 'Find an ECT' page.
- [x]       5a. If user goes back via the 'check details' link, the fields on the 'Find an ECT' page will be reset to blank and the user can enter new details to find a different person's TRS record.
- [ ] 6. A magic TRN and date of birth or TRS test data is provided for testing the 'Find an ECT' page with a TRS match. 
- [ ]       6a. The magic or test data should be shown somewhere on the page for information when testing the journey (i.e. in review / test / QA environments, not production). Example below.
- [ ] 7. There will be a 'Back' link on each page that takes the user back to the previous screen in the journey.

![image](https://github.com/user-attachments/assets/afc89051-e861-4117-99f7-9614dea06429)

## Technical notes

Validation on TRN and date of birth will be added in a separate ticket pending #457. 

Eligibility criteria for who can be added as an ECT will also be added in a separate later ticket. @peteryates mentioned there is something reusable from AB portal work for unhappy paths for eligibility.

Note, the prototype version of the 'Check if this is your ECT' page includes an additional question that is **out of scope** for this ticket (see below).

![Image](https://github.com/user-attachments/assets/e6b21fef-2da3-4e69-9988-4e611c3131a3)

## Linked tickets
Dependent on #468

**Comment by gracecbrennan** [comment]
Making good progress on this - hoping to have first one ready today.

**Comment by mason-emily** [comment]
@TobyRet I've added a link to this spreadsheet which has the URLs for each page - please use those rather than what's in the prototype

**Comment by peteryates** [comment]
> Eligibility criteria for who can be added as an ECT will also be added in a separate later ticket. @peteryates mentioned there is something reusable from AB portal work for unhappy paths for eligibility.

Some of them will be shared between ABP and register ECTs, like when the teacher isn't found we raise a `TRS::Errors::TeacherNotFound` which you'd be able to intercept and display an appropriate error message here.

**Comment by TobyRet** [comment]
Part 1 has been merged - https://github.com/DFE-Digital/register-early-career-teachers/pull/530 

Now working on part 2 - calling the TRS API and returning teacher data based on submitted `trn` and `date of birth`

**Comment by claire-hughez** [comment]
Blocked till Toby is back

**Comment by gracecbrennan** [comment]
Some small changes on AC - aiming to have a PR out today.

**Comment by gracecbrennan** [comment]
Toby has had some feedback from Claire and Tony and have a few more changes to push. This should be complete soon
